src/AccessibilityInsights.Actions/Actions/ListenAction.cs

```diff
@@ -42,73 +36,43 @@ public class ListenAction:IDisposable
         /// <param name="config"></param>
         /// <param name="ec"></param>
         /// <param name="listener"></param>
-        private ListenAction(RecorderSetting config, ElementContext ec, HandleUIAutomationEventMessage listener)
+        private ListenAction(TreeScope listenScope, ElementContext ec, HandleUIAutomationEventMessage listener)
         {
             this.Id = Guid.NewGuid();
-            this.Config = config;
             this.ElementContext = ec;
-            this.EventListener = new EventListenerFactory(ec.Element, config.ListenScope);
-            this.EventRecords = new List<IA11yEventMessage>();
+            this.EventListener = new EventListenerFactory(ec.Element, listenScope);
             this.ExternalListener = listener;
         }
 
         /// <summary>
-        /// Start recoding events
+        /// Start recording events
         /// </summary>
-        public void Start()
+        public void Start(IEnumerable<int> eventIDs, IEnumerable<int> propertyIDs)
         {
-            this.EventRecords.Clear();
             this.IsRunning = true;
-            InitFocusChangedEventListener();
-            InitIndividualEventListeners();
-            InitPropertyChangeListener();
+            InitIndividualEventListeners(eventIDs);
+            InitPropertyChangeListener(propertyIDs);
         }
 
         /// <summary>
         /// Start Property change Event listener
         /// </summary>
-        private void InitPropertyChangeListener()
+        private void InitPropertyChangeListener(IEnumerable<int> propertyIds)
         {
-            var list = from c in this.Config.Properties
-                       where this.Config.IsListeningAllEvents || c.CheckedCount > 0
-                       select c.Id;
-
-            if (list.Count() != 0)
+            if (propertyIds.Any())
             {
-                this.EventListener.RegisterAutomationEventListener(EventType.UIA_AutomationPropertyChangedEventId, this.onEventFired, list.ToArray());
+                this.EventListener.RegisterAutomationEventListener(EventType.UIA_AutomationPropertyChangedEventId, this.onEventFired, propertyIds.ToArray());
             }
         }
 
         /// <summary>
         /// Start Individual Event Listener
         /// </summary>
-        private void InitIndividualEventListeners()
+        private void InitIndividualEventListeners(IEnumerable<int> eventIds)
         {
-            var list = from c in this.Config.Events
-                       where this.Config.IsListeningAllEvents || c.CheckedCount > 0
-                       select c;
-
-            foreach (var l in list)
+            foreach (var id in eventIds)
             {
-                if (l.Id == EventType.UIA_StructureChangedEventId)
```

Comment:
flagging - I combined the if/else here, please let me know if this could cause any issues; the prior structure seems to be deliberate